lib/ex_unit/lib/ex_unit/filters.ex

```diff
@@ -14,51 +16,63 @@ defmodule ExUnit.Filters do
   on the command line) includes a line number filter, and if so returns the
   appropriate ExUnit configuration options.
   """
-  @spec parse_path(String.t()) :: {String.t(), Keyword.t()}
-  def parse_path(file) do
-    case extract_line_numbers(file) do
-      {path, []} -> {path, []}
-      {path, line_numbers} -> {path, exclude: [:test], include: line_numbers}
-    end
+  @spec parse_path(String.t()) :: {String.t(), ex_unit_opts}
+  def parse_path(file_path) do
+    {[parsed_path], ex_unit_opts} = parse_paths([file_path])
+    {parsed_path, ex_unit_opts}
   end
 
-  defp extract_line_numbers(file) do
-    case String.split(file, ":") do
-      [part] ->
-        {part, []}
-
-      parts ->
-        {reversed_line_numbers, reversed_path_parts} =
-          parts
-          |> Enum.reverse()
-          |> Enum.split_while(&match?({_, ""}, Integer.parse(&1)))
-
-        line_numbers =
-          for line_number <- reversed_line_numbers,
-              valid_line_number?(line_number),
-              reduce: [],
-              do: (acc -> [line: line_number] ++ acc)
-
-        path =
-          reversed_path_parts
-          |> Enum.reverse()
-          |> Enum.join(":")
-
-        {path, line_numbers}
-    end
+  @doc """
+  Like `parse_path/1` but for multiple paths.
+
+  ExUnit filter options are combined.
+  """
+  @spec parse_paths([String.t()]) :: {[String.t()], ex_unit_opts}
+  def parse_paths(file_paths) do
+    {parsed_paths, locations} =
+      Enum.map_reduce(file_paths, [], fn file_path, locations ->
+        case extract_line_numbers(file_path) do
+          {path, []} -> {path, locations}
+          {path, [line]} -> {path, [{:location, {path, line}} | locations]}
+          {path, lines} -> {path, [{:location, {path, lines}} | locations]}
+        end
```

Comment:
I wanted to add these changes as well. I think it's much nicer if the `:location` tuple just contains all the given line numbers. :grinning: